app_dart/lib/src/service/datastore.dart

```diff
@@ -55,6 +55,18 @@ class DatastoreService {
     return query.run();
   }
 
+  // Queries for recent commits without considering branches.
+  // TODO(keyonghan): combine this function with the above `queryRecentCommits`,
+  // this needs to fix https://github.com/flutter/flutter/issues/52694.
+  Stream<Commit> queryRecentCommitsNoBranch({int limit = 100, int timestamp}) {
```

Comment:
Are 100 commits enough? I remember that Casey once extended the memcache from 3 months of commits to 6 months of commits to properly triage some performance issues. If there are 10 commits a day, that's a limit of 1800 commits.

Reply:
That's the default number for this function. The actual limit for history API is 6000.
https://github.com/flutter/cocoon/blob/c2bf4d635f65dc9203e621a23504ec9b35eeeebf/app_dart/lib/src/request_handlers/get_timeseries_history.dart#L42

Reply:
I see. LGTM!

Reply:
To clarify, the issue @liyuqian : There was an update to the Datastore records to use a new index for the branching release strategy in Flutter. This an unintended side effect as Cocoon never saved previous records with a branch until the past few weeks. This adds a new query that doesn't use the branch field (at the moment there are very few timeseries datapoints that will impact the perf graphs). 

The long term solution is going to be to update old timeseries data to have a master branch (as Cocoon only processed data on master before).

Reply:
Thanks @CaseyHillers for the explanation. And thank you @keyonghan for a quick fix!